airbyte-webapp/src/pages/connectorBuilder/ConnectorBuilderLandingPage/ConnectorBuilderLandingPage.tsx

```diff
@@ -0,0 +1,214 @@
+      const reader = new FileReader();
+      reader.onload = async (readerEvent) => {
+        const yaml = readerEvent.target?.result as string;
+        const fileName = file?.name;
+
+        try {
+          let json;
+          try {
+            json = load(yaml) as ConnectorManifest;
+          } catch (e) {
+            if (e instanceof YAMLException) {
+              registerNotification({
+                id: YAML_UPLOAD_ERROR_ID,
+                text: (
+                  <FormattedMessage
+                    id={YAML_UPLOAD_ERROR_ID}
+                    values={{
+                      reason: e.reason,
+                      line: e.mark.line,
+                    }}
+                  />
+                ),
+                type: ToastType.ERROR,
+              });
+            }
+            return;
+          }
+
+          let convertedFormValues;
+          try {
+            convertedFormValues = await convertToBuilderFormValues(json, DEFAULT_BUILDER_FORM_VALUES);
+          } catch (e) {
+            setStoredEditorView("yaml");
+            setStoredManifest(json);
+            navigate(ConnectorBuilderRoutePaths.Edit);
+            return;
+          }
+
+          if (fileName) {
+            convertedFormValues.global.connectorName = startCase(fileName.split(".")[0]);
+          }
+          setStoredEditorView("ui");
+          setStoredFormValues(convertedFormValues);
+          navigate(ConnectorBuilderRoutePaths.Edit);
+        } finally {
+          if (fileInputRef.current) {
+            fileInputRef.current.value = "";
+          }
+          setImportYamlLoading(false);
+        }
+      };
+
+      if (file) {
+        reader.readAsText(file);
+      }
+    },
+    [
+      convertToBuilderFormValues,
+      navigate,
+      registerNotification,
+      setStoredEditorView,
+      setStoredFormValues,
+      setStoredManifest,
+    ]
+  );
+
+  return (
+    <FlexContainer direction="column" alignItems="center" gap="2xl">
+      <FlexContainer direction="column" gap="md" alignItems="center" className={styles.titleContainer}>
+        <AirbyteLogo />
+        <Heading as="h1" size="lg" className={styles.title}>
+          <FormattedMessage id="connectorBuilder.landingPage.title" />
+        </Heading>
+      </FlexContainer>
+      <Heading as="h1" size="lg">
+        <FormattedMessage id="connectorBuilder.landingPage.prompt" />
+      </Heading>
+      <FlexContainer direction="row" gap="2xl">
+        <input type="file" accept=".yml,.yaml" ref={fileInputRef} onChange={handleYamlUpload} hidden />
+        <Tile
+          image={<ImportYamlImage />}
+          title="connectorBuilder.landingPage.importYaml.title"
+          description="connectorBuilder.landingPage.importYaml.description"
+          buttonText="connectorBuilder.landingPage.importYaml.button"
+          buttonProps={{ isLoading: importYamlLoading }}
+          onClick={() => {
+            unregisterNotificationById(YAML_UPLOAD_ERROR_ID);
```

Comment:
We should always unregister this when navigating away from the page, otherwise the user could try to upload a yaml, give up and go with the "from scratch" option:
<img width="1131" alt="Screenshot 2023-02-02 at 15 35 27" src="https://user-images.githubusercontent.com/1508364/216353735-8d526c10-ca54-479e-8186-b9f8789937f9.png">

Can be done via return value of an effect hook
```

useEffect(() => {
  return () => unregisterNotificationById(YAML_UPLOAD_ERROR_ID);
}, [unregisterNotificationById]);
```

Reply:
Good catch! Fixed